solana/solana-ibc/programs/solana-ibc/src/storage.rs

```diff
@@ -129,13 +129,41 @@ pub(crate) struct PrivateStorage {
     pub ibc_events_history: BTreeMap<InnerHeight, Vec<InnerIbcEvent>>,
 }
 
+/// Provable storage, i.e. the trie, held in an account.
+pub type AccountTrie<'a, 'b> =
+    solana_trie::AccountTrie<core::cell::RefMut<'a, &'b mut [u8]>>;
+
+/// Checks contents of given unchecked account and returns a trie if it’s valid.
+///
+/// The account needs to be owned by [`crate::ID`] and
+pub fn get_provable_from<'a, 'info>(
+    info: &'a UncheckedAccount<'info>,
+    name: &str,
+) -> Result<AccountTrie<'a, 'info>> {
+    fn get<'a, 'info>(
```

Comment:
Why do we have this code block as a method when we are using it only once?

Reply:
It’s going to be used in other places in the future.

Still, even though it’s used once, it’s a good unit of code to extract.

Reply:
Alright, makes sense.